node/staking_test.go

```diff
@@ -40,12 +40,16 @@ func TestUpdateStakingList(t *testing.T) {
 	node := New(host, consensus, testDBFactory, false)
 	node.BlockPeriod = 8 * time.Second
 
-	for i := 0; i < 5; i++ {
+	for i := 0; i < 1; i++ {
 		selectedTxs := node.getTransactionsForNewBlock(common.Address{})
 		node.Worker.CommitTransactions(selectedTxs, common.Address{})
-		block, _ := node.Worker.Commit([]byte{}, []byte{}, 0, common.Address{})
+		block, err := node.Worker.FinalizeNewBlock([]byte{}, []byte{}, 0, common.Address{}, nil, nil)
 
-		err := node.AddNewBlock(block)
+		if err != nil {
+			t.Errorf("Error when finalizing block: %v", err)
+		}
+		block.Header()
```

Comment:
Are you adding this to finalize the header and compute the hash?

Reply:
yea, before, the finalization may fail, then the test will have nil pointer error.

Reply:
Makes sense, can you add a comment to explain that we are actually calling this accessor to rely upon its side effect?  The line looks very out of place.

Reply:
Done.